src/Analyzer/Passes/LogicalExpressionOptimizerPass.cpp

```diff
@@ -93,47 +186,71 @@ class JoinOnLogicalExpressionOptimizerVisitor : public InDepthQueryTreeVisitorWi
           *     b => [(a IS NULL AND b IS NULL)]
           *     c => [(a IS NULL AND c IS NULL)]
           * }
-          * Then for each a <=> b we can find all operands that contains both a IS NULL and b IS NULL
+          * Then for each equality a = b we can check if we have operand (a IS NULL AND b IS NULL)
           */
         QueryTreeNodePtrWithHashMap<std::vector<size_t>> is_null_argument_to_indices;
 
+        bool is_anything_changed = false;
+
         for (const auto & argument : function_node.getArguments())
         {
-            or_operands.push_back(argument);
+            if (isNodeBooleanConstant(argument, false))
```

Comment:
Should we also optimize for true? Any true found here should automatically allow ignoring any other conditions (and even remove the or if we wanted to and do a cross join).

Reply:
An example:
```
EXPLAIN QUERY TREE
SELECT *
FROM t1
INNER JOIN t2 ON (t1.x = t2.x) OR ((t1.x IS NULL) AND (t2.x IS NULL)) OR (t1.x = 5)
ORDER BY t1.x ASC NULLS LAST
```

Works fine. Now if you replace (`t1.x = 5`) with `true`, nothing is optimized

Reply:
In case we have `... OR true` the whole join expression is replaced with constant in some (another optimization) and we execute cross join. So, I suppose, we do not need to handle this case here.